src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -58,9 +58,14 @@ internal ItemManager(RecentItemsManager recentItemsManager, IGlobalOptionService
         private static SegmentedList<VSCompletionItem> SortCompletionitems(AsyncCompletionSessionInitialDataSnapshot data, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
-            var items = new SegmentedList<VSCompletionItem>(data.InitialItemList);
-            items.Sort(VSItemComparer.Instance);
+            var items = new SegmentedList<VSCompletionItem>(data.InitialItemList.Count);
+            foreach (var item in data.InitialItemList)
+            {
+                CompletionItemData.GetOrAddDummyRoslynItem(item);
```

Comment:
We don't expect problems because we're modifying the properties of completion items that aren't our own?

Reply:
No, the property bag is supposed to be used by LanguageServices for whatever purposes they see fit.